dockerfiles/Dockerfile-src-rosetta

```diff
@@ -0,0 +1,170 @@
+RUN mkdir --mode=700 ~/.gnupg && \
+  echo 'export PATH="$PATH:/usr/lib/go/bin:$HOME/.cargo/bin"' >> ~/.bashrc && \
+  echo 'ulimit -s unlimited' >> ~/.bashrc && \
+  echo "disable-ipv6" >> ~/.gnupg/dirmngr.conf
+
+
+# bubblewrap was disabled in other builds via the dockerfile-toolchain images, and in the default opam2 image.
+# Importantly, this also allows the entire container to be built with a permissionless builder, avoiding docker-in-docker
+# Keeping the exact install steps from the official opam image in case we want to re-enable in the future.
+#RUN curl -fL https://github.com/projectatomic/bubblewrap/releases/download/v0.4.1/bubblewrap-0.4.1.tar.xz | \
+#   tar -xJ && \
+#   cd bubblewrap-0.4.1 && ./configure --prefix=/usr/local && make && sudo make install && \
+#   cd - && rm -rf bubblewrap-0.4.1
+
+
+# Avoid the need to pass -y everywhere, although we still do because its not hard
+ENV OPAMYES 1
+
+# Ocaml install of a given OCAML_VERSION via opam switch
+# additionally initializes opam with sandboxing disabled, as we did not install bubblewrap above.
+RUN git clone git://github.com/ocaml/opam-repository /home/opam/opam-repository && \
+  opam init --disable-sandboxing -k git -a ~/opam-repository --bare && \
+  opam switch create "${OCAML_VERSION}" "ocaml-base-compiler.${OCAML_VERSION}${OCAML_REVISION}" && \
+  opam switch "${OCAML_VERSION}"
+
+# Alternate variants for 4.07 that are included in the official opam image
+#  opam switch create 4.07+afl ocaml-variants.4.07.1+afl && \
+#  opam switch create 4.07+flambda ocaml-variants.4.07.1+flambda && \
+#  opam switch create 4.07+default-unsafe-string ocaml-variants.4.07.1+default-unsafe-string && \
+#  opam switch create 4.07+force-safe-string ocaml-variants.4.07.1+force-safe-string && \
+
+
+#################################################################################################
+# The "opam-deps" Stage
+# - Continues from the build-deps image
+# - Installs all opam dependencies and pins from coda's github
+# - Includes the entire coda codebase and submodules in "${CODA_DIR}"
+# - Largely mirrors/replaces ./scripts/setup-opam.sh
+#################################################################################################
+FROM build-deps AS opam-deps
+
+# location of repo used for pins and external package commits
+ARG CODA_DIR=/coda
+
+# location of external packages
+ARG EXTERNAL_PKG_DIR=$CODA_DIR/src/external
+
+# don't keep sources, to force reinstall of pinned packages from Coda sources
+# and to keep Docker image reasonable size
+ARG OPAMKEEPBUILDDIR=false
+ARG OPAMREUSEBUILDDIR=false
+
+RUN git clone https://github.com/CodaProtocol/coda "${CODA_DIR}" && git checkout rosetta/dockerfile
+WORKDIR ${CODA_DIR}
+RUN git submodule update --init --recursive
+
+# All our ocaml packages
+RUN opam switch import src/opam.export && eval $(opam config env)
+
+# Extlib gets automatically installed, but we want our pin, so we should
+# uninstall here
+RUN opam uninstall extlib
```

Comment:
This is a little concerning: won't it uninstall any packages that depend on `extlib` too? I think you should be able to pin over it, like we do with `async_kernel`, and it'll recompile as needed.

Reply:
It definitely does not do anything destructive, but I'm glad you pointed it out because its actually a side effect of the opam2 docker container we had been using previously (when the container is built, the 1 and only installed opam package is extlib), but in our environment extlib is not actually installed by default with opam, so in this case it just prints "Nothing to do. [NOTE] extlib is not installed."

Given that its a no-op with questionable value to begin with, I'm removing this line.